framework/packages/abstract-core/src/native/ibc_host.rs

```diff
@@ -0,0 +1,167 @@
+        pub client_proxy_address: String,
+        pub account_id: AccountId,
+        pub action: HostAction,
+        pub chain_name: ChainName,
+    }
+}
+/// Used by Abstract to instantiate the contract
+/// The contract is then registered on the version control contract using [`crate::version_control::ExecuteMsg::ProposeModules`].
+#[cosmwasm_schema::cw_serde]
+pub struct InstantiateMsg {
+    /// Used to easily perform address translation on the app chain
+    pub ans_host_address: String,
+    /// Used to create remote abstract accounts
+    pub account_factory_address: String,
+    /// Version control address
+    pub version_control_address: String,
+}
+
+#[cosmwasm_schema::cw_serde]
+pub struct MigrateMsg {}
+
+#[cosmwasm_schema::cw_serde]
+#[non_exhaustive]
+pub enum InternalAction {
+    /// Registers a new account from a remote chain
+    Register {
+        name: String,
+        description: Option<String>,
+        link: Option<String>,
+    },
+}
+
+#[cosmwasm_schema::cw_serde]
+#[non_exhaustive]
+pub enum HelperAction {
+    /// What do we need here ? TODO
+    SendAllBack,
+}
+
+/// Callable actions on a remote host
+#[cosmwasm_schema::cw_serde]
+pub enum HostAction {
+    Dispatch {
+        manager_msg: manager::ExecuteMsg,
+    },
+    /// Can't be called by an account directly. These are permissionned messages that only the IBC Client is allowed to call by itself.
+    Internal(InternalAction),
+    /// Some helpers that allow calling dispatch messages faster (for actions that are called regularly)
+    Helpers(HelperAction),
+}
+
+/// Interface to the Host.
+#[cosmwasm_schema::cw_serde]
+#[cfg_attr(feature = "interface", derive(cw_orch::ExecuteFns))]
+pub enum ExecuteMsg {
+    /// Update the Admin
+    UpdateAdmin { admin: String },
+    UpdateConfig {
+        ans_host_address: Option<String>,
+        account_factory_address: Option<String>,
+        version_control_address: Option<String>,
+    },
+    /// Register the Polytone proxy for a specific chain.
+    /// proxy should be a local address (will be validated)
+    RegisterChainProxy { chain: String, proxy: String },
+    /// Remove the Polytone proxy for a specific chain.
+    RemoveChainProxy { chain: String },
+    /// Create an account internally (used for account creation before account action)
+    InternalRegisterAccount {
+        client_chain: String,
+        account_id: AccountId,
+    },
+    /// Allow for fund recovery through the Admin
+    RecoverAccount {
+        closed_channel: String,
+        account_id: AccountId,
+        msgs: Vec<CosmosMsg>,
+    },
+    /// Allows for remote execution from the Polytone implementation
+    #[cfg_attr(feature = "interface", fn_name("ibc_execute"))]
+    Execute {
+        account_id: AccountId,
+        /// The address of the calling account id. This is used purely for the send-all-back method.
+        /// We include it in all messages one-the-less to simpify the users life
+        proxy_address: String,
```

Comment:
If this is only used for send_all_back, can we store it in the host somehow?

Reply:
No, the idea is that accountIds don't change, so we can denote accounts using those. However, proxy addresses may change. So we can't store that information and trust it

Reply:
Because of the possibility of chain resets? I.e. uni-5 -> uni-6 ?

Reply:
Yes or for whatever reason the proxy address might change. It doesn't seem reasonable to put a constraint on the proxy address when the account already has much constraints on it

Reply:
I don't think we'll ever want to enable a re-mapping of account-id to manager/proxy.

Reply:
Ok, so we store the proxy address in the ibc host ?